### ISSUE TITLE
Sign of hyperfine constant breaks sorting of states

### ISSUE THREAD

**Issue by RHanley1** [OxfordIonTrapGroup/atomic_physics#45]
I have come across a peculiar issue with the sorting/indexing of states. If one uses a positive hyperfine constant value, then it appears that the indexing of the states no longer works correctly. 

For example, lets look at the ca43 example code. I would expect that a change in the sign of Ahfs and gI just inverts the Breit-Rabi diagram, and this is indeed what one observed if using the breit_rabi.py example script. However, if we use an example script which relies on indexing the states, such as the clock_qubits.py example code, then it fails. This is because the index ```ion.index(ca43.ground_level, -4, F=4)``` returns ```array([], shape=(0, 1), dtype=int64)```.  This is not, at least to me, the expected behaviour, 

I have tried to dig into this but am not having much luck. Is there something I am missing here, as I would naively expect the indexing to work for any sign of Ahfs?

**Comment by hartytp** [comment]
Thanks @RHanley1 ! You're right this is a clear bug in the code. I think that @sotirova has seen the same bug recently and was planning to open an issue as well (*cough cough* :) )

**Comment by RHanley1** [comment]
Glad to hear I'm not going mad! Do you have an inclination into the cause of this - it wasn't obvious to me? I'm happy to take a look at a fix

**Comment by hartytp** [comment]
I'm afraid I don't really other than to point out that I never tested with positive hyperfine values so this is perhaps maybe not too unexpected in hindsight.

Stating the obvious I have probably accidentally made some assumption that was true for 43Ca+ but is not generally true. Most likely in the sorting code. If you're able to have a look I'd really appreciate that.

**Comment by hartytp** [comment]
What would be really helpful though is a minimal failing test case (e.g. make a test atom data structure with positive AHFs etc)

**Comment by sotirova** [comment]
PR coming in a few minutes!

**Comment by hartytp** [comment]
🎉

**Comment by sotirova** [comment]
I belive `b32491c` fixes this.

**Comment by hartytp** [comment]
Yes, thanks! But let's keep this open until we have a regression test
